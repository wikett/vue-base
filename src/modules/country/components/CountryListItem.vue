<script setup lang="ts">
import type { Country } from '@/modules/country/services/types'
import { GlobeEuropeAfricaIcon, UserGroupIcon } from '@heroicons/vue/20/solid'
const props = defineProps<{
  country: Country
}>()
</script>
<template>
  <li
    :key="props.country.name.official"
    class="col-span-1 my-1 flex flex-col divide-y divide-gray-200 rounded-lg bg-transparent text-center shadow shadow-cyan-700 transition-opacity duration-100 ease-out"
  >
    <div class="flex flex-1 flex-col px-8 pt-8">
      <img
        class="mx-auto h-32 w-full flex-shrink-0 rounded-md object-contain"
        :src="props.country.flags.svg"
        :alt="props.country.flags.alt"
      />
      <h3 class="mt-6 text-sm font-semibold text-secondary-800">
        {{ props.country.name.official }}
      </h3>
      <dl class="mt-1 flex flex-grow flex-col justify-between">
        <dd v-if="props.country.capital" class="mt-3">
          <span
            class="inline-flex items-center rounded-full bg-green-50 px-2 py-1 text-xs font-medium text-green-700 ring-1 ring-inset ring-green-600/20"
            >{{ props.country.capital[0] }}</span
          >
        </dd>
      </dl>
    </div>
    <div>
      <div class="-mt-px flex divide-x divide-gray-200">
        <div class="flex w-0 flex-1">
          <div
            class="relative inline-flex w-0 flex-1 items-center justify-center gap-x-3 rounded-br-lg border border-transparent py-4 text-sm font-semibold text-gray-900"
          >
            <div class="flex flex-col items-center">
              <GlobeEuropeAfricaIcon class="h-5 w-5 text-gray-400" aria-hidden="true" />
              <p class="text-sm font-semibold">{{ props.country.continents[0] }}</p>
            </div>
          </div>
        </div>
        <div class="-ml-px flex w-0 flex-1">
          <div
            class="relative inline-flex w-0 flex-1 items-center justify-center gap-x-3 rounded-br-lg border border-transparent py-4 text-sm font-semibold text-gray-900"
          >
            <div class="flex flex-col items-center">
              <UserGroupIcon class="h-5 w-5 text-gray-400" aria-hidden="true" />
              <p>{{ props.country.population.toLocaleString() }}</p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </li>
</template>
