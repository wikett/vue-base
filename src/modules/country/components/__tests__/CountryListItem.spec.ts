import { describe, it, expect } from 'vitest'
import { mount } from '@vue/test-utils'
import CountryListItem from '@/modules/country/components/CountryListItem.vue'
import { countryOne } from '@/modules/country/components/__tests__/fakeCountry'

describe('CountryListItem', () => {
  it('should render correctly', () => {
    const wrapper = mount(CountryListItem, {
      props: {
        country: countryOne
      }
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
