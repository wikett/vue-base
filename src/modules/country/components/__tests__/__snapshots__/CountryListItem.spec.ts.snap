// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CountryListItem > should render correctly 1`] = `
"<li class="col-span-1 my-1 flex flex-col divide-y divide-gray-200 rounded-lg bg-transparent text-center shadow shadow-cyan-700 transition-opacity duration-100 ease-out">
  <div class="flex flex-1 flex-col px-8 pt-8"><img class="mx-auto h-32 w-full flex-shrink-0 rounded-md object-contain" src="" alt="">
    <h3 class="mt-6 text-sm font-semibold text-secondary-800">Fake Country</h3>
    <dl class="mt-1 flex flex-grow flex-col justify-between">
      <dd class="mt-3"><span class="inline-flex items-center rounded-full bg-green-50 px-2 py-1 text-xs font-medium text-green-700 ring-1 ring-inset ring-green-600/20"></span></dd>
    </dl>
  </div>
  <div>
    <div class="-mt-px flex divide-x divide-gray-200">
      <div class="flex w-0 flex-1">
        <div class="relative inline-flex w-0 flex-1 items-center justify-center gap-x-3 rounded-br-lg border border-transparent py-4 text-sm font-semibold text-gray-900">
          <div class="flex flex-col items-center"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 20 20" fill="currentColor" aria-hidden="true" data-slot="icon" class="h-5 w-5 text-gray-400">
              <path fill-rule="evenodd" d="M18 10a8 8 0 1 1-16 0 8 8 0 0 1 16 0Zm-1.503.204A6.5 6.5 0 1 1 7.95 3.83L6.927 5.62a1.453 1.453 0 0 0 1.91 2.02l.175-.087a.5.5 0 0 1 .224-.053h.146a.5.5 0 0 1 .447.724l-.028.055a.4.4 0 0 1-.357.221h-.502a2.26 2.26 0 0 0-1.88 1.006l-.044.066a2.099 2.099 0 0 0 1.085 3.156.58.58 0 0 1 .397.547v1.05a1.175 1.175 0 0 0 2.093.734l1.611-2.014c.192-.24.296-.536.296-.842 0-.316.128-.624.353-.85a1.363 1.363 0 0 0 .173-1.716l-.464-.696a.369.369 0 0 1 .527-.499l.343.257c.316.237.738.275 1.091.098a.586.586 0 0 1 .677.11l1.297 1.297Z" clip-rule="evenodd"></path>
            </svg>
            <p class="text-sm font-semibold"></p>
          </div>
        </div>
      </div>
      <div class="-ml-px flex w-0 flex-1">
        <div class="relative inline-flex w-0 flex-1 items-center justify-center gap-x-3 rounded-br-lg border border-transparent py-4 text-sm font-semibold text-gray-900">
          <div class="flex flex-col items-center"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 20 20" fill="currentColor" aria-hidden="true" data-slot="icon" class="h-5 w-5 text-gray-400">
              <path d="M10 9a3 3 0 1 0 0-6 3 3 0 0 0 0 6ZM6 8a2 2 0 1 1-4 0 2 2 0 0 1 4 0ZM1.49 15.326a.78.78 0 0 1-.358-.442 3 3 0 0 1 4.308-3.516 6.484 6.484 0 0 0-1.905 3.959c-.023.222-.014.442.025.654a4.97 4.97 0 0 1-2.07-.655ZM16.44 15.98a4.97 4.97 0 0 0 2.07-.654.78.78 0 0 0 .357-.442 3 3 0 0 0-4.308-3.517 6.484 6.484 0 0 1 1.907 3.96 2.32 2.32 0 0 1-.026.654ZM18 8a2 2 0 1 1-4 0 2 2 0 0 1 4 0ZM5.304 16.19a.844.844 0 0 1-.277-.71 5 5 0 0 1 9.947 0 .843.843 0 0 1-.277.71A6.975 6.975 0 0 1 10 18a6.974 6.974 0 0 1-4.696-1.81Z"></path>
            </svg>
            <p>0</p>
          </div>
        </div>
      </div>
    </div>
  </div>
</li>"
`;
